Wrong `HsBindgen.Runtime.ConstantArray.withPtr` type signature

I found this by generating bindings for the following code:

```c
int process_buffer(char buffer[static 64], size_t size);
```

This generated the following Haskell code:

```haskell
foreign import ccall safe "Generated_process_buffer" process_buffer_wrapper :: (F.Ptr FC.CChar) -> HsBindgen.Runtime.Prelude.CSize -> IO FC.CInt

process_buffer :: ((HsBindgen.Runtime.ConstantArray.ConstantArray 64) FC.CChar) -> HsBindgen.Runtime.Prelude.CSize -> IO FC.CInt
process_buffer =
  \x0 ->
    \x1 ->
      HsBindgen.Runtime.ConstantArray.withPtr x0 (\ptr2 ->
                                                    process_buffer_wrapper ptr2 x1)
```

And when compiling, it failed type checking with the following error:

```
    • Couldn't match type ‘HsBindgen.Runtime.Prelude.CChar’
                     with ‘HsBindgen.Runtime.Prelude.CInt’
        arising from a use of ‘HsBindgen.Runtime.ConstantArray.withPtr’
    • In the expression:
        HsBindgen.Runtime.ConstantArray.withPtr
          x0 (\ ptr2 -> process_buffer_wrapper ptr2 x1)
      In the expression:
        \ x1
          -> HsBindgen.Runtime.ConstantArray.withPtr
               x0 (\ ptr2 -> process_buffer_wrapper ptr2 x1)
      In the expression:
        \ x0
          -> \ x1
               -> HsBindgen.Runtime.ConstantArray.withPtr
                    x0 (\ ptr2 -> process_buffer_wrapper ptr2 x1)
    |
670 |       HsBindgen.Runtime.ConstantArray.withPtr x0 (\ptr2 ->
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Doxygen.hs:671:76: error: [GHC-83865]
    • Couldn't match type ‘HsBindgen.Runtime.Prelude.CInt’
                     with ‘HsBindgen.Runtime.Prelude.CChar’
      Expected: F.Ptr HsBindgen.Runtime.Prelude.CChar
        Actual: F.Ptr HsBindgen.Runtime.Prelude.CInt
    • In the first argument of ‘process_buffer_wrapper’, namely ‘ptr2’
      In the expression: process_buffer_wrapper ptr2 x1
      In the second argument of ‘HsBindgen.Runtime.ConstantArray.withPtr’, namely
        ‘(\ ptr2 -> process_buffer_wrapper ptr2 x1)’
    |
671 |                                                     process_buffer_wrapper ptr2 x1)
```

Here's HLS information:

<img width="1010" height="457" alt="Image" src="https://github.com/user-attachments/assets/3e56c060-f275-4a33-bad0-06cde4e9b7cf" />

Given the above, we can see that the problem is `withPtr` type signature is wrong and shouldn't restrict the output type.

--- a/hs-bindgen-runtime/src/HsBindgen/Runtime/ConstantArray.hs
+++ b/hs-bindgen-runtime/src/HsBindgen/Runtime/ConstantArray.hs
@@ -94,7 +94,7 @@ toList (CA v) = VS.toList v
 -- Coercible abstraction to look through the `newtype`s wrappers of typedefs.
 withPtr ::
      (Coercible b (ConstantArray n a), Storable a)
-  => b -> (Ptr a -> IO a) -> IO a
+  => b -> (Ptr a -> IO r) -> IO r
 withPtr (coerce -> CA v) k = do
     -- we copy the data, a e.g. int fun(int xs[3]) may mutate it.
     VS.MVector _ fptr <- VS.thaw v

